internal/api/grpc/project/v2beta/project_metadata.go

```diff
@@ -0,0 +1,59 @@
+package project
+
+import (
+	"context"
+
+	"connectrpc.com/connect"
+	metadata "github.com/zitadel/zitadel/internal/api/grpc/metadata/v2beta"
+	filter "github.com/zitadel/zitadel/pkg/grpc/filter/v2beta"
+	project_pb "github.com/zitadel/zitadel/pkg/grpc/project/v2beta"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func (s *Server) SetProjectMetadata(
```

Comment:
If you're able, also have a look at the integration tests for user metadata under `internal/api/grpc/user/v2/integration_test/metadata_test.go`